PLANET-6147: Fix missing and breaking classes on blocks

lithrel opened the pull request:
Ref: https://jira.greenpeace.org/browse/PLANET-6147

The style name extraction method generates issues with the "additional classes" field.
Additional classes are also not included in the resulting blocks, except when the current bug extracts a style and if those additional classes are placed after the style class name.

Currently, for a Columns block with style `tasks`, an additional class `is-style-tasks` is included. If we edit the additional class to `foo is-style-tasks bar`, the column will end up with classes `block-style-tasks bar`. This is because the code splits the classname string on `is-style-`, keeps everything that is after it (so `tasks bar`) and concatenate to `block-style-`. 
Other blocks simply don't have these additional CSS classes included on their container section.

## Example

You should look at the `class` attribute of the `section` tag (usually).
**Config**:
`foobar is-style-no_image foobaz`
![Screenshot from 2021-06-17 18-22-17](https://user-images.githubusercontent.com/617346/122436298-ff4ac180-cf98-11eb-8943-1ff0514dfc87.png)
**Before fix**:
`block columns-block block-style-no_image foobaz`
![Screenshot from 2021-06-17 18-24-25](https://user-images.githubusercontent.com/617346/122436648-489b1100-cf99-11eb-9c9a-fa8d6ac08fab.png)
**After fix**:
`block columns-block block-style-no_image foobar is-style-no_image foobaz`
![Screenshot from 2021-06-17 18-28-30](https://user-images.githubusercontent.com/617346/122437267-d8d95600-cf99-11eb-9ccf-d72693372e74.png)


## Fix 

- Adding a function `getStyleFromClassName` to use in relevant blocks
- Adding className string to generated blocks containers that are missing it

This PR does not fix
- ServerSideRendered blocks : OldENForm, OldCarouselHeader, SocialMedia, TakeActionBoxout
- SplitTwoColumns (not sure I understand where `className` is) 

## Test

Here is the boring part:
- Add one of every Planet4 block in a page
- Add additional classes in the "Additional CSS class(es)" field everywhere you can
- Check that it appears in the source inspector on backend and frontend 

The less boring way:
I've edited a page on Sinope (https://www-dev.greenpeace.org/test-sinope/?page_id=47006&preview=true), adding classes `foobar` to blocks, and `foobaz` when a class `is-style-*` is involved (to check that classes included before and after this string are properly added). So you can also:
- Go there and open the inspector
- Search (ctrl+f) for classes `.foobar`, `.foobaz`
- Check that they are in there, and that classes `is-style-*` are also included

planet-4 commented:
### Test instance is ready :rocket:

:new_moon: [sinope](https://www-dev.greenpeace.org/test-sinope)

:watch: 2021.06.17 13:37:41

mleray reviewed (commented):
For the CarouselHeader, you could probably also solve this by adding the className to the WYSIWYG [version](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/master/assets/src/blocks/CarouselHeader/CarouselHeaderFrontend.js#L76)?

mleray reviewed (commented):
For the SplitTwoColumns block, I'm pretty sure that the issue is due to the filtering that happens [here](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/master/classes/blocks/class-splittwocolumns.php#L162), since `className` isn't set in the [ATTRIBUTES](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/master/classes/blocks/class-splittwocolumns.php#L36) it's removed and therefore not passed to the frontend 🤔

lithrel commented:
> For the CarouselHeader, you could probably also solve this by adding the className to the WYSIWYG [version](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/master/assets/src/blocks/CarouselHeader/CarouselHeaderFrontend.js#L76)?

@mleray Good point, missed it because of missing rebase, I just added it :+1: 

> For the SplitTwoColumns block, I'm pretty sure that the issue is due to the filtering that happens [here](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/master/classes/blocks/class-splittwocolumns.php#L162), since `className` isn't set in the [ATTRIBUTES](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/master/classes/blocks/class-splittwocolumns.php#L36) it's removed and therefore not passed to the frontend thinking

:eyes: great find :+1:  added a short fix for it

Also added a [short fix](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/616/files#diff-7b24535513d85372ec7dcdd75454546d4ca1a95d7ae7120f6fbd62a19ff82237R33) for the very rare/temporary case of adding an empty carousel header, the absence of `slides` was breaking the editor, if you have a better version I'll take it

mleray reviewed (approved):
Tested on local, looks good to me 👍
